ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionReplacementByFeeMarketRule.java

```diff
@@ -50,40 +48,38 @@ public boolean shouldReplace(
     }
 
     Wei newEffPrice = priceOf(newTransactionInfo.getTransaction(), baseFee);
-    Long newEffPriority =
-        newTransactionInfo.getTransaction().getEffectivePriorityFeePerGas(baseFee);
+    Wei newEffPriority = newTransactionInfo.getTransaction().getEffectivePriorityFeePerGas(baseFee);
 
     // bail early if price is or priority is not strictly positive
-    if (!(newEffPrice.toLong() > 0L)) {
+    if (newEffPrice.compareTo(Wei.ZERO) <= 0) {
```

Comment:
Wei is unsigned, so I think this check is superfluous

Reply:
I guess == 0 is still valid though

Reply:
simplified to only check if the value is 0